### PR TITLE
Network training and loss function

### PR DESCRIPTION
**Pull request by lvrcek**
Implemented training with custom hyperparameters and plotting the training/validation loss. Refactored code and added new unit tests. Updated demo jupyter notebook.

### CODE REVIEW

File: reactome_gnn/train.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    learning_rate = hyperparams['lr']
+    batch_size = hyperparams['batch_size']
+    device = hyperparams['device']
+
+    model_path = os.path.join(data_path, 'models')
+    model_path = os.path.join(model_path, f'model_dim{dim_latent}_lay{num_layers}.pth')
+    
+    # Create datasets
+    ds = dataset.PathwayDataset(data_path)
+    ds_train, ds_valid = [ds[0], ds[1], ds[2]], [ds[3], ds[4]]
```
Comment (teslajoy) [inline reactome_gnn/train.py]: It's interesting that the train and validation have the same curve even with this line of code.

**Review by teslajoy — APPROVED** [review]
Merge to master when you're ready. Thank you!

File: reactome_gnn/train.py (hunk 1 above)
Comment (lvrcek) [reply reactome_gnn/train.py]: Yes, this is a bit "hardcoded", but simply because there is not a lot of graphs in this toy dataset, which is also the reason for the same curves. With more graphs in the dataset we could simply split the dataset randomly into train/validation, and with more graphs in both train/validation dataset I'm sure the losses would be more noisy. I could try to do that in the next few days.

**Comment by teslajoy** [comment]
Awesome, only if u have time. No pressure 

All the best,
_Nasim

**Comment by lvrcek** [comment]
Created a bit larger dataset, updated the demo, did some basic hyperparameter optimization. You can check the notebook now for the updated loss functions. Still similar, but you can notice the differences.
